gui/default/syncthing/core/syncthingController.js

```diff
@@ -11,15 +11,28 @@ angular.module('syncthing.core')
         var navigatingAway = false;
         var online = false;
         var restarting = false;
+        var authenticated = !!window.metadata;
```

Comment:
Would this be clear enough? :)
```suggestion
        var authenticated = Boolean(window.metadata);
```

Reply:
Oh I am not talking about the `!!`, that's clear. To me it's not obvious how window.metadata being present means authenticated.

Reply:
Ah, ok. It's because `window.metadata` is set in the script `/meta.js` which is behind the authentication requirement, so `window.metadata` won't be set (because the script fails to load) if we're not yet authenticated.

I could add a comment here explaining that, but I'm thinking perhaps it would make sense to move that implementation detail to the backend. We could change this to `var authenticated = window.metadata && window.metadata.authenticated;` and always set `window.metadata.authenticated = true` in the backend (with the comment there instead, explaining why it's always true).

Which would you prefer? An explanatory comment in the frontend, or a new `metadata` attribute encapsulating the implementation details with a explanatory comment in the backend instead?

Reply:
Ah no that feels redundant if it's anyway only availabe when authed - just a comment seems appropriate (and/or util function if this is used in another place too)

Reply:
Fixed in 5feda50caf0d5e2234fca4469a323e3f20bc0ee6.